lib/nexthop_group.c

```diff
@@ -592,31 +617,60 @@ static bool nexthop_group_parse_nexthop(struct nexthop *nhop,
 	} else
 		nhop->type = NEXTHOP_TYPE_IFINDEX;
 
+	if (labels) {
+		int ret;
+		uint8_t num = 0;
+		mpls_label_t larray[MPLS_MAX_LABELS];
+
+		ret = mpls_str2label(labels, &num, larray);
+		if (ret == 0 && num > 0) {
+			nexthop_add_labels(nhop, ZEBRA_LSP_NONE,
+					   num, larray);
+		}
+	}
+
 	return true;
 }
 
+/*
+ * Wrapper to parse the strings in a 'nexthop_hold'
+ */
+static bool nexthop_group_parse_nhh(struct nexthop *nhop,
+				    const struct nexthop_hold *nhh)
+{
+	return (nexthop_group_parse_nexthop(nhop, nhh->addr,
+					    nhh->intf,
+					    nhh->nhvrf_name,
+					    nhh->labels));
+}
+
 DEFPY(ecmp_nexthops, ecmp_nexthops_cmd,
       "[no] nexthop\
         <\
 	  <A.B.C.D|X:X::X:X>$addr [INTERFACE$intf]\
 	  |INTERFACE$intf\
 	>\
-	[nexthop-vrf NAME$vrf_name]",
+	[{ \
+	   nexthop-vrf NAME$vrf_name \
+	   |label WORD \
+	}]",
       NO_STR
       "Specify one of the nexthops in this ECMP group\n"
       "v4 Address\n"
       "v6 Address\n"
       "Interface to use\n"
       "Interface to use\n"
       "If the nexthop is in a different vrf tell us\n"
-      "The nexthop-vrf Name\n")
+      "The nexthop-vrf Name\n"
+      MPLS_LABEL_HELPSTR)
```

Comment:
nit: here's the definition of `MPLS_LABEL_HELPSTR`:
```c
#define MPLS_LABEL_HELPSTR                                                     \
        "Specify label(s) for this route\nOne or more "                        \
        "labels in the range (16-1048575) separated by '/'\n"
```

In this case we're specifying the label(s) of a nexthop and not a route...

Reply:
Yes, got it - I was just trying to reuse something, but you are right, better not to confuse anyone